enterprise/backend/test/metabase_enterprise/sandbox/api/dashboard_test.clj

```diff
@@ -14,6 +15,46 @@
    [schema.core :as s]
    [toucan2.core :as t2]))
 
+(deftest params-values-test
```

Comment:
Dan:

A lot of times these tests can be flaky and really tricky because sometimes the field values already exist and sometimes they don't. I added

```clojure
(t2/delete! 'FieldValues :field_id
            [:in [(mt/id :users :name)
                  (mt/id :categories :name)
                  (mt/id :venues :name)]])
```
Right before the assertion and I got a failure:

```clojure
expected: {46100 {:values ["Garaje" "Gordo Taqueria" "La Tortilla"], :human_readable_values [], :field_id 46100},
           46112 {:values ["African" "American" "Artisan"], :human_readable_values [], :field_id 46112}}
  actual: {46100 {:values ("Garaje" "Gordo Taqueria" "La Tortilla"), :human_readable_values (), :field_id 46100}}
    diff: - {46112 {:values ["African" "American" "Artisan"], :human_readable_values [], :field_id 46112}}
```

Do you know what should happen here if none of these field values exist? 

I'm sorry to harp so much on this but this is a source of really hard to track down bugs. If the answer is "if they exist the user can see them" then that's fine if a bit weird. But at least it is deliberate. And if so we should test them that way.

https://github.com/metabase/metabase/pull/29359#discussion_r1170027893

Reply:
this is basically related to [this](https://github.com/metabase/metabase/pull/30217/files#r1170878780) problem.

When you call delete like that, it deletes all full and advanced FieldValues for a field. this And since the field `(mt/id :categories :name)` is not sandboxed in this case, it calls to the OSS implementation and get the full values(not sandboxed values). And because the current implementation of full values does not `get-or-create`, we got nothing back.